### PR TITLE
Feature/lumberjack toggle planting

### LINKED ISSUE #639: SUGGESTION: Allow lumberjack to not plant seedlings.
Have a way to enable / disable the placement of seedlings by the lumberjack, after cutting down a tree, to allow for the clearing of an area to build in.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/42236815-suggestion-allow-lumberjack-to-not-plant-seedlings?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by justin-munyer**
Closes #639 

# Changes proposed in this pull request:
- Allow toggling of lumberjack sapling replanting, to allow for tree clearing.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java [hunk 1]
```diff
@@ -85,6 +92,26 @@ public WindowHutLumberjack(final BuildingLumberjack.View building)
...
+        MineColonies.getNetwork().sendToServer(new LumberjackReplantSaplingToggleMessage(building, ownBuilding.shouldReplant));
+        updateReplantButton();
+    }
+
+    private void updateReplantButton()
+    {
+        Button buttonReplant = findPaneOfTypeByID(BUTTON_TOGGLE_REPLANT, Button.class);
+
+        if (ownBuilding.shouldReplant) {buttonReplant.setLabel(LanguageHandler.format(TOGGLE_REPLANT_SAPLINGS_ON));}
+        else buttonReplant.setLabel(LanguageHandler.format(TOGGLE_REPLANT_SAPLINGS_OFF));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java]: CRITICAL Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java [hunk 2]
```diff
@@ -11,8 +11,10 @@
 import com.minecolonies.blockout.views.SwitchView;
 import com.minecolonies.coremod.MineColonies;
 import com.minecolonies.coremod.colony.buildings.BuildingLumberjack;
+import com.minecolonies.coremod.network.messages.LumberjackReplantSaplingToggleMessage;
 import com.minecolonies.coremod.network.messages.LumberjackSaplingSelectorMessage;
 import net.minecraft.item.ItemStack;
+import org.apache.commons.codec.language.bm.Lang;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java]: MINOR Remove this unused import 'org.apache.commons.codec.language.bm.Lang'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java [hunk 3]
```diff
@@ -85,6 +92,26 @@ public WindowHutLumberjack(final BuildingLumberjack.View building)
...
+        ownBuilding.shouldReplant = !ownBuilding.shouldReplant;
+        MineColonies.getNetwork().sendToServer(new LumberjackReplantSaplingToggleMessage(building, ownBuilding.shouldReplant));
+        updateReplantButton();
+    }
+
+    private void updateReplantButton()
+    {
+        Button buttonReplant = findPaneOfTypeByID(BUTTON_TOGGLE_REPLANT, Button.class);
+
+        if (ownBuilding.shouldReplant) {buttonReplant.setLabel(LanguageHandler.format(TOGGLE_REPLANT_SAPLINGS_ON));}
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java]: MAJOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00122)
MINOR Move this left curly brace to the beginning of next line of code. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)
MINOR Make this line start at column 13. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 4]
```diff
@@ -281,6 +304,8 @@ private void checkTreesToFell()
          */
         public final Map<ItemStorage, Boolean> treesToFell = new LinkedHashMap<>();
 
+        public boolean shouldReplant = true;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: MINOR Make shouldReplant a static final constant or non-public and provide accessors if needed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 5]
```diff
@@ -254,6 +267,16 @@ public void serializeToView(@NotNull final ByteBuf buf)
         }
     }
 
+    public boolean shouldReplant(){
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 6]
```diff
@@ -341,16 +341,24 @@ private AIState chopTree()
...
+            if (((BuildingLumberjack) building).shouldReplant())
+            {
+                plantSapling();
+            }
+            else
+            {
+                job.tree = null;
+                checkedInHut = false;
+            }
+                building.getColony().getStatsManager().incrementStatistic("trees");
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java]: MINOR Make this line start at column 13. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 7]
```diff
@@ -341,16 +341,24 @@ private AIState chopTree()
             }
         }
 
-        if (!job.tree.hasLogs() && (!job.tree.isSlimeTree() || !job.tree.hasLeaves()))
-        {
-            if (hasNotDelayed(WAIT_BEFORE_SAPLING))
-            {
+        if (!job.tree.hasLogs() && (!job.tree.isSlimeTree() || !job.tree.hasLeaves())) {
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 8]
```diff
@@ -341,16 +341,24 @@ private AIState chopTree()
...
+            if (hasNotDelayed(WAIT_BEFORE_SAPLING)) {
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 62 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 51 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L92](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L92): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L300](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L300): Reduce this class from 44 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/b81bda0071d6fd0b895014c8d5abde4abd637b8e/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 9]
```diff
@@ -292,10 +318,16 @@ public View(final ColonyView c, final BlockPos l)
             super(c, l);
         }
 
+        /**
+         * Whether or not the LJ should replant saplings
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: this one seems at the wrong place.

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 10]
```diff
@@ -347,10 +347,20 @@ private AIState chopTree()
             {
                 return getState();
             }
-            plantSapling();
-            this.getOwnBuilding().getColony().getStatsManager().incrementStatistic("trees");
-            workFrom = null;
-            return LUMBERJACK_GATHERING;
+
+            final AbstractBuilding building = getOwnBuilding();
+            if (((BuildingLumberjack) building).shouldReplant())
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java]: You can do "getOwnBuilding(BuildingLumberjack.class)" as well.

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 54 issues
* ![MAJOR][MAJOR] 50 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L92](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L92): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L301](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L301): Reduce this class from 44 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 11]
```diff
@@ -254,6 +267,24 @@ public void serializeToView(@NotNull final ByteBuf buf)
...
+     */
+    public boolean shouldReplant()
+    {
+        return replant;
+    }
+
+    /**
+     * Set whether or not LJ should replant saplings
+     * @param shouldReplant
+     */
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: MINOR Make shouldReplant a static final constant or non-public and provide accessors if needed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 54 issues
* ![MAJOR][MAJOR] 50 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L92](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L92): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L308](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L308): Reduce this class from 44 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/3cf6c3a1073b7db02951240814d2ed8013ed7ae4/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 12]
```diff
@@ -281,6 +312,11 @@ private void checkTreesToFell()
          */
         public final Map<ItemStorage, Boolean> treesToFell = new LinkedHashMap<>();
 
+        /**
+         * Whether or not the LJ should replant saplings
+         */
+        public boolean shouldReplant = true;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java:318]: MINOR Make shouldReplant a static final constant or non-public and provide accessors if needed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 53 issues
* ![MAJOR][MAJOR] 50 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L92](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L92): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L308](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L308): Reduce this class from 44 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/ab35496ea03a0fd267c863d1bc1d898847d01cc1/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
Looks good :shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2485/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**